Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -634,17 +636,18 @@ void WiiSocket::Update(bool read, bool write, bool except)
           }
 
           u32 flags = memory.Read_U32(BufferIn + 0x04);
-          // Not a string, Windows requires a char* for recvfrom
-          char* data = (char*)memory.GetPointer(BufferOut);
           int data_len = BufferOutSize;
+          // Not a string, Windows requires a char* for recvfrom
+          char* data = reinterpret_cast<char*>(memory.GetPointerForRange(BufferOut, data_len));
```

Comment:
`BufferOutSize` would be better than `data_len` here so you're not converting signedness back-and-forth. (this networking code in general could probably use a good sanity checking sweep...)